src/test/java/org/elasticsearch/rest/action/admin/indices/upgrade/UpgradeReallyOldIndexTest.java

```diff
@@ -59,5 +62,6 @@ public void testUpgradeMixed_0_20_6_and_0_90_6() throws Exception {
 
         // We succeeded in upgrading only the ancient segments but leaving the "merely old" ones untouched:
         assertTrue(UpgradeTest.hasOldButNotAncientSegments(client(), indexName));
+        assertEquals(Version.CURRENT.luceneVersion.toString(), client().admin().indices().prepareGetSettings(indexName).get().getSetting(indexName, IndexMetaData.SETTING_VERSION_MINIMUM_COMPATIBLE));
     }
```

Comment:
I'm confused about this assert. especially given the comments above it. If this minimum version is a lucene version, should it not be the oldest lucene version? In this case the ancient segments will be upgraded, but the "merely old" ones will still be there, so I can't see it being CURRENT.